fe/src/main/java/org/apache/doris/load/routineload/ScheduleRule.java

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.doris.load.routineload;
+
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.common.InternalErrorCode;
+import org.apache.doris.system.SystemInfoService;
+
+public class ScheduleRule {
+
+    private static int aliveBeCount(String clusterName) {
+        SystemInfoService systemInfoService = Catalog.getCurrentSystemInfo();
+        return systemInfoService.getClusterBackendIds(clusterName, true).size();
+    }
+    /**
+     * 是否开启自动调度
+     * 只针对状态是PAUSED JOB 才需要进行判断 其他都返回false
+     * @param jobRoutine
+     * @return
+     */
+    public static boolean isNeedAutoSchedule(RoutineLoadJob jobRoutine) {
+        if (jobRoutine.state != RoutineLoadJob.JobState.PAUSED) {
+            return false;
+        }
+        if (jobRoutine.autoResumeLock) {//表示只能手动恢复
+            return false;
+        }
+        /**
+         * 这种错误是 BE挂掉才会出现的场景
+         * 增加计数器、恢复锁定功能主要是为了避免无限制的恢复
+         */
+        if (jobRoutine.pauseReason.getCode() == InternalErrorCode.REPLICA_FEW_ERR) {
+            int alive =  aliveBeCount(jobRoutine.clusterName);
+            if (alive < jobRoutine.replicationNum) {// be存活个数小于最小副本数
+                return false;
+            }
+            if (jobRoutine.firstResumeTimestamp == 0) {//说明第一次自动恢复
+                jobRoutine.firstResumeTimestamp = System.currentTimeMillis();
+                jobRoutine.autoResumeCount = 1;
+                return true;
+            } else {
+                long current = System.currentTimeMillis();
+                if (current - jobRoutine.firstResumeTimestamp < 5*60*1000) {//在5分钟之内
```

Comment:
Make the `5min` configuration. Add it to `Config.java`.
And I think we should add another config to disable this `auto resume` feature, in case we've got some bad cases.

Reply:
I add a configuration for the time